model/service/BatteryService.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017 (original work) Open Assessment Technologies SA;
+ *
+ */
+
+namespace oat\taoBattery\model\service;
+
+use oat\taoBattery\model\BatteryException;
+use oat\taoBattery\model\model\BatteryModel;
+
+interface BatteryService
+{
+    const SERVICE_ID = 'taoBattery/batteryService';
+
+    /**
+     * Construct the battery model from the given battery param
+     * Return an implementation of battery model, following the used service
+     *
+     * @param $battery
+     * @return BatteryModel
+     * @throws BatteryException
+     */
+     public function buildBattery($battery);
+
+    /**
+     * Create a battery
+     *
+     * @param $label
+     * @param $deliveries
+     * @return mixed
+     */
+     public function createBattery($label, $deliveries);
```

Comment:
We should have the type for deliveries? array in this case, I don't know if we have convention about this, but I see below that sometimes we have the type array in the params list.
`
 public function createBattery($label, array $deliveries);
`